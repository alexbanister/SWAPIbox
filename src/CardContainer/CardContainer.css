.cardContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr; }
  .cardContainer .load-more {
    display: flex;
    justify-content: center;
    width: 100%; }
