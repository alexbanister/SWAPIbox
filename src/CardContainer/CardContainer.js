import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import Button from '../Button/Button';


const CardContainer = ({ cardData, toggleFavorite } ) => {
  const allCards = cardData.map( (cardDataObj, index) => {
    return <Card key={index} cardData={cardDataObj} toggleFavorite={toggleFavorite}/>
  })
  return (
    <div className='cardContainer'>
      {allCards}
      <div className='prev-next'>
        <Button label='Prev' />
        <Button label='Next' />
      </div>
    </div>
  );
};

CardContainer.propTypes = {
  cardData: PropTypes.arrayOf(PropTypes.object)
};

export default CardContainer;
