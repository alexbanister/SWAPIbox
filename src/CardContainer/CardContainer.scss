@import "../variables.scss";

.cardContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  .load-more {
    display: flex;
    justify-content: center;
    width: 100%;
    // column-span: 3;
  }
}
