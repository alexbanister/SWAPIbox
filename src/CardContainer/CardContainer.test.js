import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer';
import {
  mockFilms,
  mockPeople,
  mockPlanets,
  mockVehicles,
  mockSpecies,
  mockPerson,
  mockPlanet,
  mockCardDataObj
} from '../_mock/mockData';

describe('CardContainer', () => {
  it('should render correctly and match the snapshot', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(< CardContainer handleLoadMore={mockFunc} cardData={[mockCardDataObj]} toggleFavorite={mockFunc} />);

    expect(wrapper).toMatchSnapshot();
  })
})

   
                   