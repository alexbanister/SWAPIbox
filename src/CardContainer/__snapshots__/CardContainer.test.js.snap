// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render correctly and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardData={
        Array [
            Object {
              "Homeworld": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "id": 1507831320587,
              "isFavorite": false,
              "name": "Luke Skywalker",
              "type": "people",
            },
          ]
    }
    handleLoadMore={[Function]}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Population": "200000",
                                    "Species": "Human",
                                    "id": 1507831320587,
                                    "isFavorite": false,
                                    "name": "Luke Skywalker",
                                    "type": "people",
                                  }
            }
            toggleFavorite={[Function]}
/>,
        ],
        <div
          className="load-more"
>
          <Button
                    label="Load More"
                    onClick={[Function]}
          />
</div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "Homeworld": "Tatooine",
            "Population": "200000",
            "Species": "Human",
            "id": 1507831320587,
            "isFavorite": false,
            "name": "Luke Skywalker",
            "type": "people",
          },
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            label="Load More"
            onClick={[Function]}
/>,
          "className": "load-more",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "label": "Load More",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Population": "200000",
                                          "Species": "Human",
                                          "id": 1507831320587,
                                          "isFavorite": false,
                                          "name": "Luke Skywalker",
                                          "type": "people",
                                        }
              }
              toggleFavorite={[Function]}
/>,
          ],
          <div
            className="load-more"
>
            <Button
                        label="Load More"
                        onClick={[Function]}
            />
</div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "id": 1507831320587,
              "isFavorite": false,
              "name": "Luke Skywalker",
              "type": "people",
            },
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              label="Load More"
              onClick={[Function]}
/>,
            "className": "load-more",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "label": "Load More",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
